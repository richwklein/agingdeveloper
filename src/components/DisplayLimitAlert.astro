---
/**
 * @name DisplayLimitAlert
 * Component used for for showing an alert when the number of articles exceeds
 * the limit of the page.
 *
 * @param {Props} props - The limit and total.
 * @return {Function} an Astro component factory function.
 *
 * @example
 * <DisplayLimit limit={maxDisplay} total={totalCount} />
 *
 * @todo Add search to message once implemented.
 */
import { Icon } from "astro-icon/components";

interface Props {
  displayed: number;
  total: number;
  class?: string;
}

const { displayed, total, class: className } = Astro.props;
const remaining = total - displayed;
---

{
  total > displayed ? (
    <div
      class:list={["flex items-center rounded-md bg-blue-100 p-3 text-blue-600", className]}
      role="alert"
    >
      <Icon name="mdi:information-outline" class:list={["mr-3"]} />
      <p>{`Maximum number of articles displayed, ${remaining} remaining.`}</p>
    </div>
  ) : null
}
