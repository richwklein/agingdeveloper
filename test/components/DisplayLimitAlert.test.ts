import DisplayLimitAlert from '@components/DisplayLimitAlert.astro'
import { experimental_AstroContainer as AstroContainer } from 'astro/container'
import { beforeAll, expect, test } from 'vitest'

type RenderOptions = { total?: number; className?: string }
let alert: string

beforeAll(async () => {
  alert = await render()
})

const render = async ({ total = 15, className }: RenderOptions = {}) => {
  const container = await AstroContainer.create()
  return await container.renderToString(DisplayLimitAlert, {
    props: {
      displayed: 10,
      total: total,
      class: className,
    },
  })
}
test('that it contains alert role', async () => {
  expect(alert).toContain('role="alert"')
})

test('that it contains the info icon', async () => {
  expect(alert).toContain('mdi:information-outline')
})

test('that it contains the alert', async () => {
  expect(alert).toContain('Maximum number of articles displayed, 5 remaining.')
})

test('that it does not show when displayed is more than total', async () => {
  const result = await render({ total: 10 })
  expect(result).not.toContain('Maximum')
})

test('that class is set', async () => {
  const result = await render({ className: 'test-class' })
  expect(result).toContain('test-class')
})
