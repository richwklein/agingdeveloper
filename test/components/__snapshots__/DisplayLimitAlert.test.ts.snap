// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`DisplayLimitAlert > that it renders correctly. 1`] = `"<!DOCTYPE html><div class="flex items-center rounded-md bg-blue-100 p-3 text-blue-600" role="alert" data-astro-source-file="/Users/richklein/Development/Projects/agingdeveloper/src/components/DisplayLimitAlert.astro" data-astro-source-loc="31:6"><svg width="1em" height="1em" viewBox="0 0 24 24" class="mr-3" data-icon="mdi:information-outline" data-astro-source-file="/Users/richklein/Development/Projects/agingdeveloper/node_modules/astro-icon/components/Icon.astro" data-astro-source-loc="117:44">  <symbol id="ai:mdi:information-outline" data-astro-source-file="/Users/richklein/Development/Projects/agingdeveloper/node_modules/astro-icon/components/Icon.astro" data-astro-source-loc="124:28"><path fill="currentColor" d="M11 9h2V7h-2m1 13c-4.41 0-8-3.59-8-8s3.59-8 8-8s8 3.59 8 8s-3.59 8-8 8m0-18A10 10 0 0 0 2 12a10 10 0 0 0 10 10a10 10 0 0 0 10-10A10 10 0 0 0 12 2m-1 15h2v-6h-2z"/></symbol><use xlink:href="#ai:mdi:information-outline" data-astro-source-file="/Users/richklein/Development/Projects/agingdeveloper/node_modules/astro-icon/components/Icon.astro" data-astro-source-loc="125:10"></use>  </svg><p data-astro-source-file="/Users/richklein/Development/Projects/agingdeveloper/src/components/DisplayLimitAlert.astro" data-astro-source-loc="33:10">Maximum number of articles displayed, 5 remaining.</p></div>"`;
